lib/ui/painting/image_descriptor.cc

```diff
@@ -85,17 +85,21 @@ void ImageDescriptor::initRaw(Dart_Handle descriptor_handle,
                               int row_bytes,
                               PixelFormat pixel_format) {
   SkColorType color_type = kUnknown_SkColorType;
+  SkAlphaType alpha_type = kPremul_SkAlphaType;
   switch (pixel_format) {
     case PixelFormat::kRGBA8888:
       color_type = kRGBA_8888_SkColorType;
       break;
     case PixelFormat::kBGRA8888:
       color_type = kBGRA_8888_SkColorType;
       break;
+    case PixelFormat::kRGBAFloat32:
+      color_type = kRGBA_F32_SkColorType;
+      alpha_type = kUnpremul_SkAlphaType;
```

Comment:
Why are these unpremul but the other ones are premul?

Reply:
From the Dart API perspective it's not the only one that uses unpremultiplied, `rawStraightRgba` also returns unpremultiplied alpha.  I'm not sure what flow of the code that goes through to achieve it, but I'm maintaining the old behavior here for old flows.  I chose to make this unpremultiplied since that is the most expected result and the benefits of using premultiplied alpha don't exist here since we have to do a conversion anyway because we can't use f32 directly, we have to convert to f16.

Reply:
Consider adding a comment about that.

Reply:
I got a bit confused between `toByteData` and `decodeImageFromPixels`.  The input to toByteData was documented as non-premultiplied, but the input to decodeImageFromPixels wasn't.  I've updated all the documentation for `ui.PixelFormat` to be explicit about pixel format.  I also added a comment in this code.  The important thing is I wanted `ImageByteFormat.rawExtendedRgba128` and `PixelFormat.rgbaFloat32` to match so people can easily download, tweak, then reupload.